gguf-py/scripts/gguf-new-metadata.py

```diff
@@ -101,7 +101,7 @@ def copy_with_new_metadata(reader: gguf.GGUFReader, writer: gguf.GGUFWriter, new
 
     for tensor in reader.tensors:
         # Dimensions are written in reverse order, so flip them first
-        shape = np.flipud(tensor.shape)
+        shape = np.flipud(tensor.shape).tolist()
```

Comment:
Any particular reason for coercing shape to list here (and not elsewhere)?

Reply:
It's passed to `GGUFWriter.add_tensor_info`, which expects a `Sequence[int]` for the shape, and this shape is of type `NDArray[uint32]` which caused the error:

```
Argument of type "NDArray[uint32]" cannot be assigned to parameter "tensor_shape" of type "Sequence[int]" in function "add_tensor_info"
"NDArray[uint32]" is incompatible with "Sequence[int]"
(reportGeneralTypeIssues)
```

Explicitly coercing it into a list fixes the error.

Reply:
I see, although `GGUFWriter.add_tensor_info`s typing is then perhaps not correct I understand why it's simpler to just make it a list.

Reply:
`GGUFWriter.add_tensor_info`'s typing seems correct to me; it's used in 2 other places, and both use shapes which are already compatible with `Sequence[int]`.

https://github.com/ggerganov/llama.cpp/blob/b0d943de179ad5dbd83d51f327fb566066f4ccda/gguf-py/gguf/gguf_writer.py#L247-L248

https://github.com/ggerganov/llama.cpp/blob/b0d943de179ad5dbd83d51f327fb566066f4ccda/convert.py#L1145

So using `Sequence[int]` there seems appropriate, as it's the most general type (I think?) which can be indexed (it avoids having to cast `tuple[int, ...]` into `list[int]`, or `list[int]` into `tuple[int, ...]`).
This is how the shape is used in `add_tensor_info`:

https://github.com/ggerganov/llama.cpp/blob/b0d943de179ad5dbd83d51f327fb566066f4ccda/gguf-py/gguf/gguf_writer.py#L210-L211

Reply:
Sure, all I'm saying is that that also works with NDArray (even though it's not compatible with Sequence).